packages/cli/src/networks/index.ts

```diff
@@ -136,12 +143,25 @@ export function enrsToNetworkConfig(enrs: string[]): RecursivePartial<IBeaconNod
  */
 export async function fetchWeakSubjectivityState(
   config: IChainForkConfig,
-  url: string
-): Promise<TreeBacked<allForks.BeaconState>> {
+  {weakSubjectivityServerUrl, weakSubjectivityCheckpoint}: WeakSubjectivityFetchOptions
+): Promise<{wsState: TreeBacked<allForks.BeaconState>; wsCheckpoint: Checkpoint}> {
   try {
-    const response = await got(url, {headers: {accept: "application/octet-stream"}});
+    let wsCheckpoint;
+    if (weakSubjectivityCheckpoint) {
+      wsCheckpoint = getCheckpointFromArg(weakSubjectivityCheckpoint);
+    } else {
+      wsCheckpoint = await fetchFinalizedCheckpoint(
+        `${weakSubjectivityServerUrl}/eth/v1/beacon/states/finalized/finality_checkpoints`
+      );
+    }
+    const stateSlot = wsCheckpoint.epoch * SLOTS_PER_EPOCH;
+    const apiVersion = config.getForkName(stateSlot) === ForkName.phase0 ? "v1" : "v2";
```

Comment:
Should you use the lodestar-api client for this? Since we already have it, and exposes both v1 and v2 returning bytes